plugins/modules/ndo_pod_profile.py

```diff
@@ -0,0 +1,353 @@
+        pod_settings=dict(type="str", aliases=["pod_settings_name"]),
+        pod_settings_uuid=dict(type="str"),
+        pod_settings_template=dict(type="str"),
+        pod_settings_template_id=dict(type="str"),
+        state=dict(type="str", default="query", choices=["absent", "query", "present"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        mutually_exclusive=[
+            ("template", "template_id"),
+            ("pod_settings", "pod_settings_uuid"),
+            ("pod_settings_template", "pod_settings_template_id"),
+        ],
+        required_if=[
+            ["state", "absent", ["name", "uuid"], True],
+            ["state", "present", ["name", "uuid"], True],
+            ["state", "present", ["pod_settings", "pod_settings_uuid"], True],
+            ["state", "present", ["pod_settings_template", "pod_settings_template_id"], True],
+        ],
+        required_one_of=[
+            ["template", "template_id"],
+        ],
+    )
+
+    mso = MSOModule(module)
+    mso_templates = MSOTemplates(mso)
+
+    template_name = module.params.get("template")
+    template_id = module.params.get("template_id")
+    name = module.params.get("name")
+    uuid = module.params.get("uuid")
+    description = module.params.get("description")
+    pods = module.params.get("pods")
+    pod_settings = module.params.get("pod_settings")
+    pod_settings_uuid = module.params.get("pod_settings_uuid")
+    pod_settings_template = module.params.get("pod_settings_template")
+    pod_settings_template_id = module.params.get("pod_settings_template_id")
+    state = module.params.get("state")
+
+    mso_template = mso_templates.get_template("fabric_resource", template_name, template_id)
+    mso_template.validate_template("fabricResource")
+
+    match = mso_template.get_pod_profile_object(uuid, name)
+
+    if (uuid or name) and match:
+        set_pod_profile_policy_names(mso, mso_templates, match.details)
+        mso.existing = mso.previous = copy.deepcopy(match.details)  # Query a specific object
+    elif match:
+        mso.existing = [set_pod_profile_policy_names(mso, mso_templates, obj) for obj in match]  # Query all objects
+
+    pod_profile_path = "/fabricResourceTemplate/template/podProfiles/{0}".format(match.index if match else "-")
+
+    ops = []
+
+    if state == "present":
+
+        mso_values = {
+            "name": name,
+            "description": description,
+        }
+
+        if pods:
+            mso_values["blocks"] = pods
+            mso_values["kind"] = "podRange"
+        elif pods == []:
+            mso_values["kind"] = "all"
+
+        pod_settings_mso_template = mso_templates.get_template("fabric_policy", pod_settings_template, pod_settings_template_id)
+        mso_values["policy"] = pod_settings_mso_template.get_pod_settings_object(pod_settings_uuid, pod_settings, fail_module=True).details.get("uuid")
```

Comment:
Can we ignore the pod settings object lookup when the "pod_settings_uuid" is provided by the user?